### PR TITLE
feat: customize useHomeStorage

### PR DESCRIPTION
**Pull request by mario-mui**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/home/src/components/CustomHomepage/types.ts [hunk 1]
```diff
@@ -98,6 +98,14 @@ export type CustomHomepageGridProps = {
    * @defaultValue false
    */
   preventCollision?: boolean;
+
+  /**
+   * Hook that overwrite default useHomeStorage hook. For example, use database to storage homepage layout data
```
Comment (emmaindal) [inline plugins/home/src/components/CustomHomepage/types.ts:103]: can this be accomplished by instead implementing a custom version of the storage api?

**Review by emmaindal — COMMENTED** [review]
Hello @mario-mui thanks for your contribution, can I ask you to explain this use case a little bit just for my understanding of this need. I've added one comment here with a potential alternative, but I'm afraid I'm missing some context. Thank you!

File: plugins/home/src/components/CustomHomepage/types.ts (hunk 1 above)
Comment (freben) [reply plugins/home/src/components/CustomHomepage/types.ts:103]: Yes, that's the intended way.

Follow along with these instructions, and then everything that uses the storage API will have it persisted across sessions.

https://github.com/backstage/backstage/tree/master/plugins/user-settings-backend

**Comment by freben** [comment]
@mario-mui Note the comment above about actually replacing the underlying storageApi instead; closing this for now

**Pull request closed without merging**